In [2]:
#Importing necessary libraries
import pandas as pd
import numpy as np

In [3]:
#Loading the datasets for data cleaning
data=pd.read_csv("C:\\Users\\USER\\OneDrive\\Documents\\Social Buzz Project\\Content.csv")

In [4]:
data.head()

Unnamed: 0                            Content ID  \
0           0  97522e57-d9ab-4bd6-97bf-c24d952602d2   
1           1  9f737e0a-3cdd-4d29-9d24-753f4e3be810   
2           2  230c4e4d-70c3-461d-b42c-ec09396efb3f   
3           3  356fff80-da4d-4785-9f43-bc1261031dc6   
4           4  01ab84dd-6364-4236-abbb-3f237db77180   

                                User ID   Type        Category  \
0  8d3cd87d-8a31-4935-9a4f-b319bfe05f31  photo        Studying   
1  beb1f34e-7870-46d6-9fc7-2e12eb83ce43  photo  healthy eating   
2  a5c65404-5894-4b87-82f2-d787cbee86b4  photo  healthy eating   
3  9fb4ce88-fac1-406c-8544-1a899cee7aaf  photo      technology   
4  e206e31b-5f85-4964-b6ea-d7ee5324def1  video            food   

                                                 URL  
0  https://socialbuzz.cdn.com/content/storage/975...  
1  https://socialbuzz.cdn.com/content/storage/9f7...  
2  https://socialbuzz.cdn.com/content/storage/230...  
3  https://socialbuzz.cdn.com/content/storage/356...  
4  https://socialbuzz.cdn.com/content/storage/01a...

In [5]:
#Checking for number of columns and rows
no_of_rows=data.shape[0]
no_of_columns=data.shape[1]
print(no_of_rows)
print(no_of_columns)

1000
6


The content dataset has 1000 rows and 6 columns.

In [7]:
#DATA CLEANING 
#HANDLING MISSING VALUES
#general view of how many missing values there is per column 
missing_values_count=data.isnull().sum()
missing_values_count[0:6]

Unnamed: 0      0
Content ID      0
User ID         0
Type            0
Category        0
URL           199
dtype: int64

The url column has 199 missing values
We are going to drop the URL column as it is not important in the analysis.

In [8]:
#dropping the entire URL column using the inplace=true function
data.drop("URL", axis=1, inplace=True)

In [9]:
#Changing the dataheads to snake_case
data.columns=data.columns.str.replace(" ","_")
print(data.columns)

Index(['Unnamed:_0', 'Content_ID', 'User_ID', 'Type', 'Category'], dtype='object')


In [23]:
#identifying the rows that have incorrect data using boolean indexing
column_name = "Category"
# Use the str.replace() method to replace values case-insensitively
data[column_name] = data[column_name].str.replace("Studying", "studying", case=False)
data[column_name] = data[column_name].str.replace("Education", "education", case=False)
data[column_name] = data[column_name].str.replace("Healthy Eating", "healthy eating", case=False)
data[column_name] = data[column_name].str.replace("Science", "science", case=False)
data[column_name] = data[column_name].str.replace("Animals", "animals", case=False)
data[column_name] = data[column_name].str.replace("Veganism", "veganism", case=False)
data[column_name] = data[column_name].str.replace("Fitness", "fitness", case=False)
data[column_name] = data[column_name].str.replace("Technology", "technology", case=False)
data[column_name] = data[column_name].str.replace("Soccer", "soccer", case=False)
data[column_name] = data[column_name].str.replace("Culture", "culture", case=False)
data[column_name] = data[column_name].str.replace("Travel", "travel", case=False)

In [25]:
#Downloading the cleaned dataset
excel_file_path=r"C:\\Users\\USER\\Downloads\\Content.xlsx"
data.to_excel(excel_file_path, index=False)

In [ ]:
#Cleaning dataset 2
